docker/test/integration/minifi/flow_serialization/Minifi_flow_yaml_serializer.py

```diff
@@ -92,21 +108,22 @@ def serialize(self, connectable, root=None, visited=None):
                     res['Connections'].append({
                         'name': str(uuid.uuid4()),
                         'source id': str(connectable.uuid),
-                        'source relationship name': conn_name,
                         'destination id': str(proc.uuid),
                         'drop empty': ("true" if proc.drop_empty_flowfiles else "false")
                     })
+                    if (str(connectable.uuid) not in [x['id'] for x in res['Funnels']]):
+                        res['Connections'][-1]['source relationship name'] = conn_name
                     if proc not in visited:
-                        self.serialize(proc, res, visited)
+                        self.serialize_node(proc, res, visited)
             else:
                 res['Connections'].append({
                     'name': str(uuid.uuid4()),
                     'source id': str(connectable.uuid),
-                    'source relationship name': conn_name,
                     'destination id': str(conn_procs.uuid)
                 })
+                if (str(connectable.uuid) not in [x['id'] for x in res['Funnels']]):
```

Comment:
Nitpicking (as it seems to be a "how to write it shorter" challenge 😄  ):
if (any(str(connectable.uuid) == x['id'] for x in res['Funnels']))

Reply:
Yeah, I like this better :) Updated in f6a40ef8ad7c2e10b048100e8279dfb00a95c14d

Reply:
Almost okay but needed to be fixed as it was checking for the opposite, fixed in bb83a0d48c8d39d3ff03082e29fabd0ee370ac5b